### ISSUE TITLE
Add clientPrincipal to platform

### ISSUE THREAD

**Issue by geoffrich** [geoffrich/svelte-adapter-azure-swa#103]
Ref: https://github.com/geoffrich/svelte-adapter-azure-swa/issues/102#issuecomment-1399579017

The adapter should parse the authentication information from the [x-ms-client-principal header](https://learn.microsoft.com/en-us/azure/static-web-apps/user-information?tabs=javascript#api-functions) and add it to RequestEvent's [platform](https://kit.svelte.dev/docs/adapters#platform-specific-context).

See the comments in #102 - this header is not always set, and the documentation should reflect that.

Once this is complete, we'll want to take a followup issue to make this information available in local dev. There's a SvelteKit issue somewhere about adapters and local dev, but for now we could provide a function to be used in the `handle` hook that would accomplish the same thing during dev.
